### PR TITLE
lpc55s69: Fix pins on lpcxpresso55s69 and minor fixes to soc device tree

### PR DESCRIPTION
**Pull request by chrta**
These issues were found during implementation of https://github.com/zephyrproject-rtos/zephyr/pull/30286 
The LPC55S69 and LPC55S28 (and their development boards) are very similar. This is not tested on hardware.

Things that are fixed here:
- some pins on the Arduino header were wrong
- some pins on the mikroBUS header were wrong
- incorrect comment in soc device tree regarding sram4
- register sizes of some peripherals were wrong

### CODE REVIEW

**Comment by galak** [comment]
LGTM

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Will wait for @agansari to hopefully look over the header GPIO pin fixes.

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi [hunk 1]
```diff
@@ -72,7 +72,6 @@
 	};
 
 	sram4: memory@20040000 {
-		/* Conencted to USB bus*/
```
Comment (agansari) [inline dts/arm/nxp/nxp_lpc55S6x_common.dtsi:75]: this has changed since the initial user manual.

**Review by agansari — APPROVED** [review]
Rechecked the pin mappings and memory registers.

**Pull request merged**